diff --git a/emscripten/pdcscrn.c b/emscripten/pdcscrn.c
index d919efd..4e9b55d 100644
--- a/emscripten/pdcscrn.c
+++ b/emscripten/pdcscrn.c
@@ -161,7 +161,7 @@ int PDC_color_content(short color, short *red, short *green, short *blue)
     int r, g, b;
 
     asm("TermGlobals.getColorString(%0);"
-       :"=r"(str) :"r"(color));
+       :"=r"(color) :"r"(str));
     sscanf(str, "#%02x%02x%02x", &r, &g, &b);
 
     *red = DIVROUND(r * 1000, 255);
@@ -181,7 +181,7 @@ int PDC_init_color(short color, short red, short green, short blue)
 
     sprintf(str, "#%02x%02x%02x", r, g, b);
     asm("TermGlobals.setColor(%0, %1);"
-       ::"r"(color), "r"(str));
+       :"=r"(color) :"r"(str));
 
     wrefresh(curscr);
 
